api/eds.proto

```diff
@@ -47,48 +47,37 @@ message EndpointDiscoveryRequest {
   Node node = 2;
 }
 
-enum HealthStatus {
-  UNKNOWN = 0;
-  HEALTHY = 1;
-  UNHEALTHY = 2;
-  // Connection draining in progress -
-  // https://aws.amazon.com/blogs/aws/elb-connection-draining-remove-instances-from-service-with-care/
-  // and
-  // https://cloud.google.com/compute/docs/load-balancing/enabling-connection-draining.
-  DRAINING = 3;
-  // This value is used by HDS Remote server. From Envoy’s perspective
-  // TIMEOUT = UNHEALTHY in case EDS returns HealthStatus.
-  TIMEOUT = 4;
-}
-
 message LbEndpoint {
-  // Endpoint host address.
-  Address address = 1;
-  // Optional health status when known and supplied by EDS server.
-  HealthStatus health_status = 2;
+  EndpointHealth endpoint_health = 1;
```

Comment:
Is there a reason to combine address and health? Health is optional and I expect majority of Envoy customers will never use it.